language-support/hs/bindings/src/DA/Ledger/Services/PackageManagementService.hs

```diff
@@ -3,25 +3,66 @@
 
 {-# LANGUAGE GADTs #-}
 
-module DA.Ledger.Services.PackageManagementService (uploadDarFile) where
+module DA.Ledger.Services.PackageManagementService (
+    listKnownPackages, PackageDetails(..),
+    uploadDarFile,
+    ) where
 
-import Com.Digitalasset.Ledger.Api.V1.Admin.PackageManagementService
-import Data.ByteString(ByteString)
+import DA.Ledger.Convert
 import DA.Ledger.GrpcWrapUtils
 import DA.Ledger.LedgerService
+import DA.Ledger.Types
+import Data.ByteString(ByteString)
+import Data.Text.Lazy (Text)
 import Network.GRPC.HighLevel.Generated
+import qualified Com.Digitalasset.Ledger.Api.V1.Admin.PackageManagementService as LL
+
+data PackageDetails = PackageDetails
+    { pid :: PackageId
+    , size :: Int
+    , knownSince :: Timestamp
+    , sourceDescription :: Text
+    } deriving (Eq,Ord,Show)
+
+listKnownPackages :: LedgerService [PackageDetails]
+listKnownPackages =
+    makeLedgerService $ \timeout config ->
+    withGRPCClient config $ \client -> do
+        service <- LL.packageManagementServiceClient client
+        let LL.PackageManagementService {packageManagementServiceListKnownPackages=rpc} = service
```

Comment:
```suggestion
        let rpc = LL.packageManagementServiceListKnownPackages service
```
I'll keep suggesting this until you tell me a reason why it's bad. :)

Reply:
Well... its clearly just a small matter of style.
But since the paint colour of this particular bike shed has been questioned, I will have to reply :)

I prefer to focus here on the record values themselves rather than the record selector functions.
This makes it clear that `service` is a package of related RPCs, of which we only use one.
By using the functional selector style, it's not obvious the value is even a record.

It's related to how I don't normally like to use `fst` and prefer pattern matching to access elems.

Reply:
I agree that it's a matter of taste. We only have this discussion because Haskell's record system sucks so much. When I see code like `let Foo x y = foo` I instinctively wonder what happens if `foo` is a `Bar ...`? Oh, there's is no `Bar ...` case. And then I'm annoyed again I have to write or read the `Foo` constructor even though it doesn't carry too much useful information. Damn Haskell records.
Even though I disagree with your particular choice of style, I'll stop bothering you about since it is indeed bike shedding.

Reply:
Thank you.

> When I see code like `let Foo x y = foo` I instinctively wonder what happens if `foo` is a `Bar`
we get a static error
```
    Pattern match(es) are non-exhaustive
    In a pattern binding: Patterns not matched: (Bar _)
```

I think the issues with Haskell's record system stem from it taking a primary view that the fields are selector-functions, rather than record pattern match syntax (aka ML style)